openmdao/core/driver.py

```diff
@@ -1687,6 +1688,289 @@ def _update_result(self, result):
+
+        This method assumes that the optimizer is in a converged state, satisfying both the
+        primal constraints as well as the optimality conditions.
+
+        The estimation of which constraints are active depends upon the feasibility tolerance
+        specified. This applies to the driver-scaled values of the constraints, and should be
+        the same as that used by the optimizer, if available.
+
+        Parameters
+        ----------
+        driver_scaling : bool
+            If False, return the Lagrange multipliers estimates in their physical units.
+            If True, return the Lagrange multiplier estimates in a driver-scaled state.
+        feas_tol : float or None
+            The feasibility tolerance under which the optimization was run. If None, attempt
+            to determine this automatically based on the specified optimizer settings.
+        use_sparse_solve : bool
+            If True, use scipy.sparse.linalg.lstsq to solve for the multipliers. Otherwise, numpy
+            will be used with dense arrays.
+
+        Returns
+        -------
+        multipliers : dict[str: ArrayLike]
+            A dictionary with an entry for each active constraint and the
+            associated Lagrange multiplier value.
+        active_info : dict[str: dict
+            A dictionary with an entry for each active constraint and its
+            active indices and bounds.
+        """
+        if not self.supports['optimization']:
+            raise NotImplementedError('Lagrange multipliers are only available for '
+                                      'drivers which support optimization.')
+
+        prob = self._problem()
+
+        obj_name = list(self._objs.keys())[0]
+        constraints = self._cons
+        des_vars = self._designvars
+
+        of_totals = {obj_name, *constraints.keys()}
+
+        active_cons, active_dvs = self._get_active_cons_and_dvs(feas_atol=feas_tol,
+                                                                feas_rtol=feas_tol)
+
+        # Active cons and vs provide the active indices in the design vars and constraints.
+        # But these design vars and constraints may themselves be indices of a larger
+        # variable.
+        totals = prob.compute_totals(list(of_totals),
+                                     list(des_vars),
+                                     driver_scaling=True)
+
+        grad_f = {inp: totals[obj_name, inp] for inp in des_vars.keys()}
+
+        n = 0
+        for inp in grad_f.keys():
+            n += grad_f[inp].size
+
+        grad_f_vec = np.zeros((n))
+        offset = 0
+        for inp in grad_f.keys():
+            inp_size = grad_f[inp].size
+            grad_f_vec[offset:offset + inp_size] = grad_f[inp]
+            offset += inp_size
+
+        actives = active_cons | active_dvs
+        n_active = np.sum(np.fromiter((len(c['indices']) for c in actives.values()), dtype=int))
```

Comment:
n_active isn't used for anything other than to bail if we don't have any active cons or desvars. Isn't it easier just to check if active_cons and active_dvs are empty dicts?